.github/workflows/validate.yml

```diff
@@ -0,0 +1,34 @@
+name: validate
+on: [push, pull_request]
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    container:
+      image: docker.io/fedora:rawhide
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      # FIXME: the following should be done statically in a custom container image
+      - name: set up dependencies
+        run: |
+          dnf install -y /usr/bin/xargs
+          scripts/testing/dependency_solver.py -brt | xargs -d '\n' dnf install -y
+          scripts/testing/dependency_solver.py --pip | xargs pip install
+
+      - name: build
+        run: |
+          ./autogen.sh
+          ./configure
+          make
+
+      - name: run unit tests
+        run: |
+          make check
+
+      - name: Upload unit test log
+        if: failure()
```

Comment:
Shouldn't we want to upload logs every time? It could be interesting from the debugging purposes.

Reply:
Sure, I can do that -- I thought let's not spam github with success logs, I also wanted to attach the coverage logs anyway, I'll look into that.

Reply:
Done.